official/docker_to_bash/generate_bash_script_from_dockerfile.py

```diff
@@ -32,61 +55,100 @@ def _get_mode():
     if "update" in sys.argv:
         return MODE_UPDATE_ONLY
     raise Exception("install or update should be passed as argument")
-        
 
 
+class _OptionException(Exception):
+    pass
+
+def _get_options():
+
+    global OPTIONS
+
+    for option in OPTIONS.keys():
+        if option in sys.argv:
+            return OPTIONS[option]
+
+    error_message = ["pass as argument the desired installation:"]
+    error_message.append("\tcore: minimal code dependencies")
+    error_message.append("\tros: core + ros and selected ros packages")
+    error_message.append("\tlaas: core + dynamic graph and other laas packages")
+    error_message.append("\tutils: tools, e.g. latex or IDEs")
+    error_message.append("\tall-code: core + laas + ros")
+    error_message.append("\tall: core + laas + ros + utils")
+
+    raise _OptionException("\n".join(error_message))
+    
+
 def _execute():
 
     ubuntu_version = _get_ubuntu_version()
 
+    # installing from scratch or
+    # just updating
     mode = _get_mode()
     if mode == MODE_FULL:
         update_only=False
     else :
         update_only=True
-    
+
+    # which set to install ? e.g. core, ros, laas
+    options = _get_options()
+        
     directory_path = _get_this_script_directory()
 
     destination_path = os.path.abspath(directory_path+os.sep+".."+os.sep+"tmp"+os.sep)
     if not os.path.isdir(destination_path):
         try : os.makedirs(destination_path)
         except Exception as e :
             raise Exception ("Failed to create "+destination_path+": "+str(e))
+
     destination_file = destination_path+os.sep+"tmp_install_file"
     if os.path.isfile(destination_file):
         try: os.remove(destination_file)
         except Exception as e :
             raise Exception ("Failed to remove "+destination_file+": "+str(e))
+ 
+    full_script = []
+        
+    for option in options:
     
-    path_to_dockerfile = directory_path+os.sep+".."+os.sep+"ubuntu_"+ubuntu_version+os.sep+"docker"+os.sep+"Dockerfile"
-    path_to_dockerfile = os.path.abspath(path_to_dockerfile)
-    if not os.path.isfile(path_to_dockerfile): 
-        raise Exception("Failed to find: "+path_to_dockerfile)
+        path_to_dockerfile = str(directory_path
+                                 +os.sep
+                                 +".."+os.sep
+                                 +"ubuntu_"+ubuntu_version
+                                 +os.sep
+                                 +"docker"+os.sep
+                                 +option+os.sep
+                                 +"Dockerfile")
```

Comment:
Using `os.path.join` might be a bit cleaner.

Reply:
+1